### PR TITLE
Fixing build Failure For Vsts

### PR DESCRIPTION
**Pull request by Anipik**
The GitSyncCommitManager was failing with error that ``` SourceRoot.SourceLinkUrl is empty```

### CODE REVIEW

**Comment by eerhardt** [comment]
FYI - @tmat

File: Directory.Build.props [hunk 1]
```diff
@@ -12,6 +12,10 @@
     <DebugSymbols>true</DebugSymbols>
   </PropertyGroup>
 
+  <ItemGroup>
+    <PackageReference Include="Microsoft.SourceLink.Vsts.Git" Version="$(MicrosoftSourceLinkVersion)" PrivateAssets="all" />
```
Comment (eerhardt) [inline Directory.Build.props]: See

https://github.com/dotnet/arcade/blob/9517895d68d0ff37c75c35f9cd9545cb9122d382/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets#L8-L14

A few comments:

1. We should probably put this in that same location.
2. Should we add this to the same condition - `Condition="'$(DotNetBuildFromSource)' != 'true'"`?
3. Can we make that `SourceLinkProvider` detection better, and then we don't even need the extra `PackageReference`?  Is there some way we are building from a VSTS repo?

Comment (tmat) [reply Directory.Build.props]: 1. I'll update the toolset. It's ok to have it here for now.
2. Yes, it should be conditional on DotNetBuildFromSource. 
3. The SourceLink packages do the detection themselves. We don't need to duplicate that.

Comment (eerhardt) [reply Directory.Build.props]: > The SourceLink packages do the detection themselves. We don't need to duplicate that.

How can the SourceLink packages do the detection themselves, if we need to do the detection in order to know which SourceLink package to include?

```
<PackageReference Include="Microsoft.SourceLink.$(SourceLinkProvider)" 
```

We need to figure out what's the right `$(SourceLinkProvider)` to even know which package should get included.

Comment (tmat) [reply Directory.Build.props]: We do not need to decide which to include. We can include both. They are designed to coexist. See https://github.com/dotnet/sourcelink/blob/master/docs/README.md#configuring-projects-with-multiple-sourcelink-providers

**Comment by Anipik** [comment]
@dotnet-bot test this please

**Comment by Anipik** [comment]
@dotnet-bot test prtest/Windows/NT/Debug

**Pull request merged**